feature/settings/src/main/java/com/google/jetpackcamera/settings/ui/SettingsComponents.kt

```diff
@@ -271,37 +272,40 @@ private fun getStabilizationStringRes(
  * ON - Both preview and video are stabilized.
  * HIGH_QUALITY - Video will be stabilized, preview might be stabilized, depending on the device.
  * OFF - Preview and video stabilization is disabled.
+ *
+ * @param supportedStabilizationMode the enabled condition for this setting.
  */
 @Composable
 fun VideoStabilizeSetting(
     currentPreviewStabilization: Stabilization,
     currentVideoStabilization: Stabilization,
+    supportedStabilizationMode: SupportedStabilizationMode,
     setVideoStabilization: (Stabilization) -> Unit,
     setPreviewStabilization: (Stabilization) -> Unit
 ) {
     BasicPopupSetting(
         title = stringResource(R.string.video_stabilization_title),
         leadingIcon = null,
-        description = stringResource(
-            id = getStabilizationStringRes(
-                previewStabilization = currentPreviewStabilization,
-                videoStabilization = currentVideoStabilization
+        enabled = supportedStabilizationMode != SupportedStabilizationMode.UNSUPPORTED,
+        description = if (supportedStabilizationMode == SupportedStabilizationMode.UNSUPPORTED) {
+            stringResource(id = R.string.stabilization_description_unsupported)
+        } else {
+            stringResource(
+                id = getStabilizationStringRes(
+                    previewStabilization = currentPreviewStabilization,
+                    videoStabilization = currentVideoStabilization
+                )
             )
-        ),
+        },
         popupContents = {
             Column(Modifier.selectableGroup()) {
-                Text(
-                    text = stringResource(id = R.string.stabilization_popup_disclaimer),
-                    fontStyle = FontStyle.Italic,
-                    color = MaterialTheme.colorScheme.onPrimaryContainer
-                )
-
                 Spacer(modifier = Modifier.height(10.dp))
 
                 // on selector
                 SingleChoiceSelector(
                     text = stringResource(id = R.string.stabilization_selector_on),
                     secondaryText = stringResource(id = R.string.stabilization_selector_on_info),
+                    enabled = supportedStabilizationMode == SupportedStabilizationMode.FULL,
```

Comment:
Do you need to do something similar for the "high quality selector" below?

Reply:
Not necessary for the UI's current implementation. if the device can't support video stabilization at all, then the popup  @won't be able to open in the first place. the whole button itself would be greyed out and unclickable.
<img width="250" alt="image" src="https://github.com/google/jetpack-camera-app/assets/20013168/399f30de-4bf7-4f50-b2b0-f23d3050439d">

Reply:
Aren't video stabilization and preview stabilization mutually exclusive?

If preview stabilization is supported, but video stabilization isn't, then you should be able to choose between `ON` and `OFF`.

Alternatively if preview stabilization isn't supported but video stabilization is, you should be able to select between `HIGH_QUALITY` and `OFF`.

Reply:
Thats true,  I assumed if the output feed couldn't be stabilized, the user may not want to bother with stabilization settings. I think to avoid confusion with `ON` meaning: "preview and video" OR "preview only if video isn't supported", I could add an extra option/enum `PREVIEW_ONLY`. 

The setting would only be completely disabled if neither preview nor video were supported.

Reply:
> I could add an extra option/enum PREVIEW_ONLY. It can be a hidden option only displayed when the device doesn't support video.

I think there's still some confusion here about preview stabilization. Preview stabilization means that Preview AND Video will be stabilized (it's WYSIWYG, hence why we call it "`ON`"). You don't need to enable video stabilization to stabilize the video, the preview stabilization will be sufficient to stabilize both. When you enable both preview and video stabilization at the same time, the preview stabilization takes precedence and stabilizes both, and the video stabilization setting on the VideoCapture use case will be ignored, hence why its ok to set it to `UNSPECIFIED`. There shouldn't be a `PREVIEW_ONLY` setting since preview stabilization will always apply a (lower quality) stabilization to the video.

> The setting would only be completely disabled if neither preview nor video were supported.

That is the correct behavior. It means it's not possible to stabilize Preview AND Video with preview stabilization. It's also not possible to get high quality video stabilization with just video stabilization.